WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.kt

```diff
@@ -0,0 +1,465 @@
+        val currentStatus = when (listType) {
+            PluginBrowserViewModel.PluginListType.SITE -> sitePluginsListStatus.value
+            PluginBrowserViewModel.PluginListType.FEATURED -> featuredPluginsListStatus.value
+            PluginBrowserViewModel.PluginListType.POPULAR -> popularPluginsListStatus.value
+            PluginBrowserViewModel.PluginListType.NEW -> newPluginsListStatus.value
+            // We should always do the initial search because the string might have changed and it is
+            // already optimized in submitSearch with a delay. Even though FluxC allows it, we don't do multiple
+            // pages of search, so if we are trying to load more, we can ignore it
+            PluginBrowserViewModel.PluginListType.SEARCH -> return !loadMore
+        }
+        if (currentStatus == PluginListStatus.FETCHING || currentStatus == PluginListStatus.LOADING_MORE) {
+            // if we are already fetching something we shouldn't start a new one. Even if we are loading more plugins
+            // and the user pulled to refresh, we don't want (or need) the 2 requests colliding
+            return false
+        }
+        return !(loadMore && currentStatus != PluginListStatus.CAN_LOAD_MORE)
+    }
+
+    fun loadMore(listType: PluginListType) {
+        fetchPlugins(listType, true)
+    }
+
+    // Network Callbacks
+
+    @Subscribe(threadMode = ThreadMode.BACKGROUND)
+    @SuppressWarnings("unused")
+    fun onWPOrgPluginFetched(event: PluginStore.OnWPOrgPluginFetched) {
+        if (event.isError) {
+            AppLog.e(T.PLUGINS, "An error occurred while fetching the wporg plugin with type: " + event.error.type)
+            return
+        }
+        // Check if the slug is empty, if not add it to the set and only trigger the update
+        // if the slug is not in the set
+        if (!TextUtils.isEmpty(event.pluginSlug) && updatedPluginSlugSet.add(event.pluginSlug)) {
+            updateAllPluginListsIfNecessary()
+        }
+    }
+
+    @Subscribe(threadMode = ThreadMode.BACKGROUND)
+    @SuppressWarnings("unused")
+    fun onPluginDirectoryFetched(event: PluginStore.OnPluginDirectoryFetched) {
+        val listStatus = if (event.isError) {
+            AppLog.e(T.PLUGINS, "An error occurred while fetching the plugin directory " + event.type + ": "
+                    + event.error.type)
+            PluginListStatus.ERROR
+        } else {
+            if (event.canLoadMore) PluginListStatus.CAN_LOAD_MORE else PluginListStatus.DONE
+        }
+        when (event.type) {
+            PluginDirectoryType.FEATURED -> _featuredPluginsListStatus.postValue(listStatus)
+            PluginDirectoryType.NEW -> _newPluginsListStatus.postValue(listStatus)
+            PluginDirectoryType.POPULAR -> _popularPluginsListStatus.postValue(listStatus)
+            PluginDirectoryType.SITE -> _sitePluginsListStatus.postValue(listStatus)
+            null -> AppLog.d(T.PLUGINS, "Plugin directory type shouldn't be null")
```

Comment:
I'll make the `event.type` `@NonNull` so we don't need to do this. It was an oversight not to do that in the first place.

Reply:
👍 It's always preferable to make types non-nullable whenever possible.